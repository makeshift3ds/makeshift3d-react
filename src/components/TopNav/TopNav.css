@import '../../assets/css/globals.css';

.top {
  background-color: var(--brown);

  & .top-container {
    display: flex;
    padding: 1rem;
    justify-content: space-between;
    max-width: var(--max-width);
    margin: 0 auto;

    @media (--small) {
      display: block;
    }
  }

  & .logo {
    height: 40px;
    width: 200px;
    background: url('../../assets/img/makeshift.svg') top left no-repeat;

    @media (--small) {
      margin: 0 auto;
    }
  }

  & .routes-nav {
    list-style-type: none;
    display: flex;
    align-items: center;
    padding: 0;
    margin: 0;
    height: 30px;

    @media (--small) {
      justify-content: center;
      margin-top: 1rem;

      & li:last-of-type a {
        margin-right: 0;
      }
    }

    & a { 
      margin-right: 1rem;
      padding: 4px;
      border-radius: 2px;
      color: var(--white);
      text-transform: uppercase;
    }

    & a:hover {
      transition: background-color 0.5s, color 0.5s;
      background-color: var(--white);
      color: var(--blue);
    }
  }
}