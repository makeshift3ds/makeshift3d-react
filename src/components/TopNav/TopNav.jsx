import React from 'react';
import './TopNav.css';

const TopNav = () => (
  <section className="top">
    <div className="top-container">
      <div className="logo" />
      <ul className="routes-nav">
        <li>
          <a href="#1" aria-label="link to the homepage">
            Home
          </a>
        </li>
        <li>
          <a href="#2" aria-label="link to the blog section">
            Blog
          </a>
        </li>
        <li>
          <a href="#3" aria-label="link to Ken Elliotts resume">
            Resume
          </a>
        </li>
        <li>
          <a href="#4" aria-label="link to the contact form">
            Contact
          </a>
        </li>
      </ul>
    </div>
  </section>
);

export default TopNav;
