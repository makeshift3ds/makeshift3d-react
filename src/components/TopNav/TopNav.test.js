import React from 'react';
import { shallow } from 'enzyme';
import TopNav from './TopNav.jsx';

describe('<TopNav />', () => {
  it('should render 1 <TopNav /> component', () => {
    const component = shallow(<TopNav />);
    expect(component).toHaveLength(1);
  });

  it('should render routes nav', () => {
    const component = shallow(<TopNav />);
    expect(component.find('ul.routes-nav')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const component = shallow(<TopNav />);
    expect(component).toMatchSnapshot();
  });
});
