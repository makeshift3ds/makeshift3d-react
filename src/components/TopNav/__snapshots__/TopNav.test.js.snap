// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="top-container"
      >
        <div
          className="logo"
        />
        <ul
          className="routes-nav"
        >
          <li>
            <Link
              aria-label="link to the homepage"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              aria-label="link to the blog section"
              replace={false}
              to="/blog"
            >
              Blog
            </Link>
          </li>
          <li>
            <Link
              aria-label="link to Ken Elliott's resume"
              replace={false}
              to="/resume"
            >
              Resume
            </Link>
          </li>
          <li>
            <a
              aria-label="link to the contact form"
              href="#4"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>,
      "className": "top",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          />,
          <ul
            className="routes-nav"
          >
            <li>
              <Link
                aria-label="link to the homepage"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                aria-label="link to the blog section"
                replace={false}
                to="/blog"
              >
                Blog
              </Link>
            </li>
            <li>
              <Link
                aria-label="link to Ken Elliott's resume"
                replace={false}
                to="/resume"
              >
                Resume
              </Link>
            </li>
            <li>
              <a
                aria-label="link to the contact form"
                href="#4"
              >
                Contact
              </a>
            </li>
          </ul>,
        ],
        "className": "top-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  aria-label="link to the homepage"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>,
              <li>
                <Link
                  aria-label="link to the blog section"
                  replace={false}
                  to="/blog"
                >
                  Blog
                </Link>
              </li>,
              <li>
                <Link
                  aria-label="link to Ken Elliott's resume"
                  replace={false}
                  to="/resume"
                >
                  Resume
                </Link>
              </li>,
              <li>
                <a
                  aria-label="link to the contact form"
                  href="#4"
                >
                  Contact
                </a>
              </li>,
            ],
            "className": "routes-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  aria-label="link to the homepage"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "link to the homepage",
                  "children": "Home",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  aria-label="link to the blog section"
                  replace={false}
                  to="/blog"
                >
                  Blog
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "link to the blog section",
                  "children": "Blog",
                  "replace": false,
                  "to": "/blog",
                },
                "ref": null,
                "rendered": "Blog",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  aria-label="link to Ken Elliott's resume"
                  replace={false}
                  to="/resume"
                >
                  Resume
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "link to Ken Elliott's resume",
                  "children": "Resume",
                  "replace": false,
                  "to": "/resume",
                },
                "ref": null,
                "rendered": "Resume",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  aria-label="link to the contact form"
                  href="#4"
                >
                  Contact
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "link to the contact form",
                  "children": "Contact",
                  "href": "#4",
                },
                "ref": null,
                "rendered": "Contact",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="top-container"
        >
          <div
            className="logo"
          />
          <ul
            className="routes-nav"
          >
            <li>
              <Link
                aria-label="link to the homepage"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                aria-label="link to the blog section"
                replace={false}
                to="/blog"
              >
                Blog
              </Link>
            </li>
            <li>
              <Link
                aria-label="link to Ken Elliott's resume"
                replace={false}
                to="/resume"
              >
                Resume
              </Link>
            </li>
            <li>
              <a
                aria-label="link to the contact form"
                href="#4"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>,
        "className": "top",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="logo"
            />,
            <ul
              className="routes-nav"
            >
              <li>
                <Link
                  aria-label="link to the homepage"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  aria-label="link to the blog section"
                  replace={false}
                  to="/blog"
                >
                  Blog
                </Link>
              </li>
              <li>
                <Link
                  aria-label="link to Ken Elliott's resume"
                  replace={false}
                  to="/resume"
                >
                  Resume
                </Link>
              </li>
              <li>
                <a
                  aria-label="link to the contact form"
                  href="#4"
                >
                  Contact
                </a>
              </li>
            </ul>,
          ],
          "className": "top-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    aria-label="link to the homepage"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>,
                <li>
                  <Link
                    aria-label="link to the blog section"
                    replace={false}
                    to="/blog"
                  >
                    Blog
                  </Link>
                </li>,
                <li>
                  <Link
                    aria-label="link to Ken Elliott's resume"
                    replace={false}
                    to="/resume"
                  >
                    Resume
                  </Link>
                </li>,
                <li>
                  <a
                    aria-label="link to the contact form"
                    href="#4"
                  >
                    Contact
                  </a>
                </li>,
              ],
              "className": "routes-nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    aria-label="link to the homepage"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "link to the homepage",
                    "children": "Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    aria-label="link to the blog section"
                    replace={false}
                    to="/blog"
                  >
                    Blog
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "link to the blog section",
                    "children": "Blog",
                    "replace": false,
                    "to": "/blog",
                  },
                  "ref": null,
                  "rendered": "Blog",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    aria-label="link to Ken Elliott's resume"
                    replace={false}
                    to="/resume"
                  >
                    Resume
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "link to Ken Elliott's resume",
                    "children": "Resume",
                    "replace": false,
                    "to": "/resume",
                  },
                  "ref": null,
                  "rendered": "Resume",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    aria-label="link to the contact form"
                    href="#4"
                  >
                    Contact
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-label": "link to the contact form",
                    "children": "Contact",
                    "href": "#4",
                  },
                  "ref": null,
                  "rendered": "Contact",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
